pkg/trace/agent/agent.go

```diff
@@ -259,6 +263,91 @@ func (a *Agent) Process(p *api.Payload, sublayerCalculator *stats.SublayerCalcul
 	}
 }
 
+var _ api.StatsProcessor = (*Agent)(nil)
+
+// ProcessStats processes incoming client stats in from the given language lang.
+func (a *Agent) ProcessStats(in pb.ClientStatsPayload, lang string) {
+	if in.Env == "" {
+		in.Env = a.conf.DefaultEnv
+	}
+	in.Env = traceutil.NormalizeTag(in.Env)
+	out := stats.Payload{
+		HostName: in.Hostname,
+		Env:      in.Env,
+	}
+	var buf strings.Builder
+	for _, group := range in.Stats {
+		for _, b := range group.Stats {
+			normalizeStatsGroup(&b, lang)
+			a.obfuscator.ObfuscateStatsGroup(&b)
+			a.Replacer.ReplaceStatsGroup(&b)
+
+			tags := map[string]string{"version": in.Version}
+			if b.HTTPStatusCode != 0 {
+				tags["http.status_code"] = strconv.Itoa(int(b.HTTPStatusCode))
+			}
+			newb := stats.Bucket{
+				Start:            int64(group.Start),
+				Duration:         int64(group.Duration),
+				Counts:           make(map[string]stats.Count),
+				Distributions:    make(map[string]stats.Distribution),
+				ErrDistributions: make(map[string]stats.Distribution),
+			}
+			grain, tagset := stats.AssembleGrain(&buf, out.Env, b.Resource, b.Service, tags)
+			key := stats.GrainKey(b.Name, stats.HITS, grain)
+			newb.Counts[key] = stats.Count{
+				Key:      key,
+				Name:     b.Name,
+				Measure:  stats.HITS,
```

Comment:
I think hits and hitsLatency misalignment is confusing. Currently 
- hits includes errors
- hitsLatencies exclude errorLatencies

I think we should either rename hitsSummary to okHitsSummary or have both exclusive

Reply:
I think `hits` is `hits` regardless of outcome. No need to rename that. However, for cases when `hits` does not include all _hits_ (excludes errors) then we should likely use the word "success" or "nonError". Then renaming to `nonErrorLatencies` is clear. WDYT?